doc/TE.md

```diff
@@ -343,6 +343,52 @@ append.  This could be used, for example to generally allow log files to be
 both writable and appendable, while allowing exceptions for certain append-only
 log scenarios.
 
+## Type casting
+
+To help protect against incorrect AV rules there are rules in place that
+restrict sources to types that have been defined as domains and targets to
+types that have been defined as resources.  If these restrictions need to be
+bypassed type casting can be used.
+
+To cast a type as to something it is not, add the type or attribute you wish
+to cast to in a set of angle brackets after using the type.
+
+```
+domain foo {
```

Comment:
The problem with this example, is that in this specific case we want people to use `self` instead of `this<resource>`.  I think we should try to have our examples point people in the right direction.  So rather than capability or process rules, I think the example should be reading /proc.